### PR TITLE
Fix: restore coinjoin accounts

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description

Stashed code during development, as usual it meant to be small but i've found some bug so i ended up refactoring some parts.

### Bug
If coinjoin account is remembered suite onload action tries to restore related CoinjoinClient and CoinjoinBackend.
If it fails CoinjoinClient instance is not created and account is in "out-of-sync" mode.
In the next account update tick will try to recreate CoinjoinBackend (and will success eventually) but CoinjoinClient will never be recreated.

### Solution
Merge `CoinjoinBackendService` and `CoinjoinClientService` as `CoinjoinService` and always initialize and return both CoinjoinClient and CoinjoinBackend.

`CoinjoinClientInstance` in coinjoin reducer receives new field `status` which works as loader since initialization of `CoinjoinService` is async.
default `CoinjoinClientInstance` is removed after unsuccessful initialization

### Commits

1. update `coinjoinAccountActions.test` code to use reduxjs/toolkit 
2. main change described above and in commit itself
3. show detailed errors in restore coinjoin session action

### CODE REVIEW

**Comment by szymonlesisz** [comment]
back to draft, i've found some edgecase where CoinjoinClient is never enabled

File: packages/suite/src/reducers/wallet/coinjoinReducer.ts [hunk 1]
```diff
@@ -219,14 +220,29 @@ const saveCheckpoint = (
...
     draft: CoinjoinState,
-    payload: ExtractActionPayload<typeof COINJOIN.CLIENT_ENABLE_SUCCESS>,
+    payload: ExtractActionPayload<typeof COINJOIN.CLIENT_ENABLE>,
 ) => {
     const exists = draft.clients[payload.symbol];
     if (exists) return;
+    draft.clients[payload.symbol] = {
+        status: 'loading',
+        rounds: [],
+        feeRatesMedians: { fast: 0, recommended: 0 },
```
Comment (szymonlesisz) [inline packages/suite/src/reducers/wallet/coinjoinReducer.ts]: To discuss: fallback from coinjoin package constants? those values should never be used its only a place holder

File: packages/suite/src/reducers/wallet/coinjoinReducer.ts [hunk 2]
```diff
@@ -219,14 +220,29 @@ const saveCheckpoint = (
...
-const createClient = (
+const initClient = (
     draft: CoinjoinState,
-    payload: ExtractActionPayload<typeof COINJOIN.CLIENT_ENABLE_SUCCESS>,
+    payload: ExtractActionPayload<typeof COINJOIN.CLIENT_ENABLE>,
 ) => {
     const exists = draft.clients[payload.symbol];
     if (exists) return;
+    draft.clients[payload.symbol] = {
+        status: 'loading',
```
Comment (szymonlesisz) [inline packages/suite/src/reducers/wallet/coinjoinReducer.ts:232]: To discuss: naming/values

Comment (marekrjpolak) [reply packages/suite/src/reducers/wallet/coinjoinReducer.ts:232]: `loading`/`loaded` sounds best imo.

**Review by marekrjpolak — APPROVED** [review]
Looks good, and I'm sure that the fixups won't make it worse! :)

File: packages/suite/src/reducers/wallet/coinjoinReducer.ts (hunk 2 above)
Comment (szymonlesisz) [reply packages/suite/src/reducers/wallet/coinjoinReducer.ts:232]: renamed, squashed, forcepushed

Comment (marekrjpolak) [reply packages/suite/src/reducers/wallet/coinjoinReducer.ts:232]: approved, approved, approved

**Pull request merged**